### PR TITLE
upgrade

### PR DESCRIPTION
**Pull request by stefanpenner**
* nearly that of ember-cli@master blueprint
* chrome over phantom
* extended test matrix
* ember-cli + other safe deps
* all deps
* remove `existsSync` -> `fs.existsSync` ended up not getting deprecated

### CODE REVIEW

**Comment by stefanpenner** [comment]
@rwjblue this is finally ready for review.

File: .travis.yml [hunk 1]
```diff
@@ -5,29 +5,43 @@ node_js:
   # so that your addon works for all apps
   - "4"
 
+sudo: false
```
Comment (scalvert) [inline .travis.yml]: Does this not need to be `sudo: required`? I've had to do this a few times lately due to the recent Meltdown/Spectre changes in travis.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,20 +1,48 @@
...
   root: true,
-  extends: ['eslint:recommended', 'prettier'],
-  plugins: ['prettier'],
   parserOptions: {
     ecmaVersion: 2017,
+    sourceType: 'module'
   },
+  plugins: [
+    'ember'
+  ],
```
Comment (scalvert) [inline .eslintrc.js]: Prettier is added as a devDep, but isn't configured. Did you intend to configure it?

File: tests/dummy/lib/ember-blog/addon/components/hello-name.js [hunk 3]
```diff
@@ -1,16 +1,21 @@
...
+    this._later = later(() => {
       if (this.isDestroyed || this.isDestroying) {
         return;
       }
       this.set('name', 'Jerry');
-    }, 500);
+    }, 50);
   },
+  destroy() {
+    cancel(this._later);
```
Comment (scalvert) [inline tests/dummy/lib/ember-blog/addon/components/hello-name.js]: Good catch.

File: .eslintrc.js (hunk 2 above)
Comment (stefanpenner) [reply .eslintrc.js]: We can configure it in a later PR.

File: .travis.yml (hunk 1 above)
Comment (stefanpenner) [reply .travis.yml]: fixed

File: node-tests/.eslintrc.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+'use strict';
```
Comment (rwjblue) [inline node-tests/.eslintrc.js]: We should be able to remove this file by adding an overrides config that matches `node-tests/**` with these settings to the top level `.eslintrc.js` file...

File: tests/dummy/lib/ember-chat/addon/views/application.js [hunk 5]
```diff
@@ -1,20 +1,5 @@
-import Ember from 'ember';
+import Component from '@ember/component';
```
Comment (rwjblue) [inline tests/dummy/lib/ember-chat/addon/views/application.js]: I think we can remove this file, we never look up `views:application` as of Ember 2.x...

**Pull request merged**